hal/src/main/native/include/hal/Notifier.h

```diff
@@ -80,13 +80,19 @@ void HAL_CancelNotifierAlarm(HAL_NotifierHandle notifierHandle,
  * Waits for the next alarm for the specific notifier.
  *
  * This is a blocking call until either the time elapses or HAL_StopNotifier
- * gets called.
+ * gets called. If the latter occurs, this function will return zero and any
+ * loops using this function should exit. Failing to do so can lead to
+ * use-after-frees.
  *
  * @param notifierHandle the notifier handle
  * @return               the FPGA time the notifier returned
  */
-uint64_t HAL_WaitForNotifierAlarm(HAL_NotifierHandle notifierHandle,
-                                  int32_t* status);
+#ifdef __cplusplus
+[[nodiscard]] uint64_t HAL_WaitForNotifierAlarm(
+#else
+uint64_t HAL_WaitForNotifierAlarm(
+#endif  // __cplusplus
+    HAL_NotifierHandle notifierHandle, int32_t* status);
```

Comment:
This is a bit ugly, you can’t wrap just the [[nodiscard]] in the ifdef and have no else?